pkg/kando/process_client.go

```diff
@@ -43,3 +54,40 @@ func processAsJSONFlagValue(cmd *cobra.Command) bool {
 	}
 	return b
 }
+
+func proxySetup(ctx context.Context, addr string, pid int64) {
+	log.Info().WithContext(ctx).Print(fmt.Sprintf("signal proxy is running for process %d", pid))
+	signalTermChan := make(chan os.Signal, 1)
+	signal.Notify(signalTermChan, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+	BREAK:
+		for {
+			select {
+			case sig := <-signalTermChan:
+				ossig, ok := sig.(syscall.Signal)
+				if !ok {
+					log.Info().WithContext(ctx).Print(fmt.Sprintf("signal %v is invalid, ignored for process %d", sig, pid))
+					continue
+				}
+				log.Info().WithContext(ctx).Print(fmt.Sprintf("signal %v received for process %d", sig, pid))
+				_, err := kanx.SignalProcess(ctx, addr, pid, int64(ossig))
+				if err != nil {
+					signal.Reset(ossig)
```

Comment:
could you please comment on why you reset here?

Reply:
In this context, I'm mostly **agnostic** on this technique as K8s reliably uses signal escalation to terminate processes.

Reply:
I can argue that it will prevent long running signal handlers from running twice (multiple SIGINTs) due to frustration, timeouts etc... this though can be addresses with a `*Once` or re-establishing the Signal handler once the signal handler has completed successfully.